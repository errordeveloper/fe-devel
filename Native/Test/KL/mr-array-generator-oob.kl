/*
 *  Copyright 2010-2012 Fabric Engine Inc. All rights reserved.
 */

operator generator(
  io Integer value
  )
{
  report("generator: value=" + value);
  value = 42;
}

operator entry() {
  ValueProducer<Size> cvg = createConstValue(Size(10));

  ArrayProducer<Integer> ag = createArrayGenerator(cvg, generator);
  report(ag);
  report(ag.getCount());
  report(ag.produce(0));
  report(ag.produce(10));
}
