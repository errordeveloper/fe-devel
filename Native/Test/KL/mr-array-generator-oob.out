ArrayProducer<Integer>
10
generator: value=0
42
Caught exception: produce index out of range
